docs/markdown/Python-module.md

```diff
@@ -238,6 +238,64 @@ module.
 
 It exposes the same methods as its parent class.
 
+## A note about Debian installation paths
+
+Debian and derivatives (Ubuntu, Linux Mint etc.) [change default installation
+and import paths][debian-dist-packages] in the system Python interpreter. The
+problem is, the way they do this is inconsistent with documented ways of
+discovering those paths inside Python. Specifically, the [`sysconfig`
+module][python-sysconfig] returns values that are not in
+[`sys.path`][python-sys-path]. This module is used internally by Meson. But it
+is generally expected that if you configure a&nbsp;build with `--prefix=/usr` or
+`--prefix=/usr/local` (the latter is the default), Python modules should be
+available without adjusting [`$PYTHONPATH`][python-pythonpath]. Therefore,
+a&nbsp;workaround is needed. There are two options that can be used in
+combination:
+
+### option to override
+
+An option in `meson_options.txt` might allow overriding the value obtained from
+[][`get_install_dir()`], and the files are installed using `install_data()`.
+This solution is very simple, but requires that all users on Debian and
+derivatives supply said option when running `meson setup`, so it's perfect when
+end users install the software from packages (and usually won't compile from
+sources).
+
+### a script to discover
+
+A variant of a following script might be used to get the path (`platlib` in this
+example):
+
+```python
+import distutils.command.install
+import distutils.sysconfig
+import distutils.util
+import pathlib
+import sys
+
+def get_platlib(prefix):
+    is_debian = 'deb_system' in distutils.command.install.INSTALL_SCHEMES
+
+    # this takes care of `/` at the end, though not `/usr/../usr/local`
+    is_usr_local = pathlib.PurePosixPath(prefix).as_posix() == '/usr/local'
+
+    if is_debian and is_usr_local:
+        return distutils.util.subst_vars(
+            distutils.command.install.INSTALL_SCHEMES['unix_local']['platlib'],
+            {
+                'platbase': '/usr',
+                'py_version_short': '.'.join(map(str, sys.version_info[:2])),
+            })
+
+    return distutils.sysconfig.get_python_lib(plat_specific=True, prefix=prefix)
```

Comment:
`distutils` is deprecated in Python 3.10, I recommend avoiding this usage.

Since this solution is still tied to the Debian patching, I think it is almost as fragile as simply hardcoding the value, and adding a sanity check to make sure it is in `sys.path`.

So, I would do
```suggestion
    try:
        sysconfig.get_paths('deb_system')
    except KeyError:
        path = '/usr/local/lib/python{}.{}/dist-packages'.format(sys.version_info[0], sys.version_info[1])
        if path not in sys.path:
            raise KeyError('Expected Debian Python platlib `{}` not in sys.path! Please open an issue.')
        return platlib
    return sysconfig.get_path('platlib')
```

Reply:
In master we already hard code the path for Debian derivatives. And we warn if it's not in sys.path. note that it will warn on Windows too. What remains to be done is to explain more in details the reason of that warning and how to avoid it by using an option to override the path.